src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,157 +3,210 @@
 import backend.resource.TurboIssue;
 import backend.resource.TurboLabel;
 import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
 import javafx.geometry.Insets;
 import javafx.scene.control.*;
-import javafx.scene.input.KeyCode;
 import javafx.scene.layout.FlowPane;
 import javafx.scene.layout.VBox;
 import javafx.stage.Modality;
 import javafx.stage.Stage;
 
-import java.util.Collections;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.ResourceBundle;
+import java.util.Set;
 import java.util.stream.Collectors;
 
-public class LabelPickerDialog extends Dialog<List<String>> {
+import org.apache.logging.log4j.Logger;
+
+import ui.UI;
+import util.HTLog;
+import util.Utility;
+
+/**
+ * Serves as a presenter that synchronizes changes in labels with dialog view  
+ * @author jinified
+ *
+ */
+public class LabelPickerDialog extends Dialog<List<String>> implements Initializable {
 
-    private static final int VBOX_SPACING = 105; // seems like some magic number
     private static final int ELEMENT_MAX_WIDTH = 400;
+    private static final Insets GROUPLESS_PAD = new Insets(5, 0, 0, 0);
+    private static final Insets GROUP_PAD = new Insets(0, 0, 10, 10);
+    private static final Logger logger = HTLog.get(LabelPickerDialog.class);
 
     private final LabelPickerUILogic uiLogic;
-    private final TextField textField;
-    private final FlowPane topPane;
-    private final VBox bottomBox;
+    private final List<TurboLabel> repoLabels;
+    private final List<String> initialLabels;
+    private final Set<String> repoLabelsSet;
```

Comment:
This is only used in one place. Maybe compute it at that point? We want to minimise the number of mutable fields.